.circleci/config.yml

```diff
@@ -1,195 +1,75 @@
-
       - store_artifacts:
           path: test-reports
           destination: test-reports
-
       - store_artifacts:
           path: coverage.xml
           destination: coverage-reports
 
-      - persist_to_workspace:
-          root: /home/circleci
-          paths:
-            - project
-
   deploy-test:
-    docker:
-      - image: cloudreach/sceptre-circleci:0.7.0
-
+    executor: python/default
     steps:
       - checkout
       - run:
-          name: Install Dependencies
-          command: |
-            python3 -m venv venv
-            . venv/bin/activate
-            pip3 install --upgrade pip
-            pip3 install -r requirements.txt
-
-          paths:
-            - 'venv'
-
-      - run:
-          name: Initialise .pypirc
+          name: Upload to PyPI
           command: |
-            echo -e "[distutils]\n  index-servers =\n    test-pypi\n\n" >> ~/.pypirc
-            echo -e "[test-pypi]" >> ~/.pypirc
-            echo -e "username: $PYPI_TEST_USER" >> ~/.pypirc
-            echo -e "password: $PYPI_TEST_PASSWORD" >> ~/.pypirc
-            echo -e "repository: https://test.pypi.org/legacy/" >> ~/.pypirc
-
-      - run:
-          name: Create Distribution
-          command: |
-            . venv/bin/activate
-            make dist
-
-      - run:
-          name: Upload to PyPi
-          command: |
-            . venv/bin/activate
-            twine upload -r test-pypi dist/*
+            poetry config repositories.test-pypi https://test.pypi.org/legacy/
+            poetry publish --build -r test-pypi -u $PYPI_TEST_USER -p $PYPI_TEST_PASSWORD
 
   deploy-production:
-    docker:
-      - image: cloudreach/sceptre-circleci:0.7.0
-
+    executor: python/default
     steps:
       - checkout
       - run:
-          name: Install Dependencies
-          command: |
-            python3 -m venv venv
-            . venv/bin/activate
-            pip3 install --upgrade pip
-            pip3 install -r requirements.txt
+          name: Upload to PyPI
+          command: poetry publish --build -u $PYPI_PROD_USER -p $PYPI_PROD_PASSWORD
 
-          paths:
-            - 'venv'
-
-      - run:
-          name: init .pypirc
-          command: |
-            echo -e "[distutils]\n  index-servers =\n    pypi\n\n" >> ~/.pypirc
-            echo -e "[pypi]" >> ~/.pypirc
-            echo -e "username: $PYPI_PROD_USER" >> ~/.pypirc
-            echo -e "password: $PYPI_PROD_PASSWORD" >> ~/.pypirc
-
-      - run:
-          name: Create Distribution
-          command: |
-            . venv/bin/activate
-            make dist
-
-      - run:
-          name: Upload to PyPi
-          command: |
-            . venv/bin/activate
-            twine upload -r pypi dist/*
 workflows:
   test:
     jobs:
       - build
-      - lint-and-unit-test:
+      - unit-tests:
           requires:
             - build
+          matrix:
+            parameters:
+              python: ["3.8", "3.9", "3.10", "3.11", "3.12"]
```

Comment:
hmm, this matrix might allow us to deprecate needing https://github.com/Sceptre/sceptre-circleci in the main sceptre repo?

Reply:
I think so, I can open a PR to sceptre with these same changes